app/controllers/authenticate_controller.rb

```diff
@@ -14,7 +14,11 @@ def oidc_authenticate_code_redirect
     ).call(
       parameters: params.to_hash.symbolize_keys,
       request_ip: request.ip
-    )
+    ) do |authenticator|
+      Authentication::AuthnOidc::V2::Strategies::AuthzCode.new(
+        authenticator: authenticator
+      ).call(params.to_hash.symbolize_keys)
+    end
```

Comment:
I'd actually recommend pushing the logic for choosing a strategy up into the `Authentication::AuthnOidc::V2::Strategy` class.  This will allow you to keep details of a particular strategy out of the controller (which ultimately should be super generic).  Something like:

```ruby
module Authentication
  module AuthnOidc
    module V2
      class Strategy
        def callback(args)
          if args.key?(:refresh_token)
            # run refresh token workflow
          elsif(args.key?(:code)
            # run code validation workflow
          end
          ...
        end
      end
    end
  end
end
```